### PR TITLE
Improve issue references in markdown

### LINKED ISSUE #2960: Qualify cross-repository ticket references
As of 0.9.13.0318 referencing a ticket of a different repository results in rendering just the target ticket number w/out the target ticket repository name. It is confusing, IMHO.
On github those kind of references are rendered as `org/repo#ticketnum` instead.

### PR DESCRIPTION
**Pull request by strk**
Ports https://github.com/gogits/gogs/pull/3436
Fixes https://github.com/gogits/gogs/issues/2960

### CODE REVIEW

**Comment by lunny** [comment]
ci failed

**Comment by lunny** [comment]
This will be back port to 1.0 but we put it on 1.1 since github cannot set two milestones on 1 issue.

**Comment by strk** [comment]
I think only setting oldest milestone is ok, assuming all bugfixex are always applied to new branches.
Beside, I don't think this is a bugfix, but rather an enhancement. Let's only put in 1.1.0, and release 1.1.0 soon (there are already many pending PR for it)

**Comment by lunny** [comment]
OK for that

**Comment by strk** [comment]
CI passes now, review please

**Comment by lunny** [comment]
LGTM

File: modules/markdown/markdown.go [hunk 1]
```diff
@@ -91,6 +91,9 @@ var (
 	IssueNumericPattern = regexp.MustCompile(`( |^|\()#[0-9]+\b`)
 	// IssueAlphanumericPattern matches string that references to an alphanumeric issue, e.g. ABC-1234
 	IssueAlphanumericPattern = regexp.MustCompile(`( |^|\()[A-Z]{1,10}-[1-9][0-9]*\b`)
+	// CrossReferenceIssueNumericPattern matches string that references a numeric issue in a difference repository
```
Comment (ethantkoenig) [inline modules/markdown/markdown.go]: different repository?

Sorry for being nit-picky, but the current version could be interpreted as if a "difference repository" is a certain type of repository.

**Comment by strk** [comment]
Thanks, fixed with 05b3f4c9217135ea2207d5a5ab68bd805c3421ea

**Comment by Bwko** [comment]
LGTM

**Pull request merged**